tests/main/hidden-snap-dir/task.yaml

```diff
@@ -0,0 +1,83 @@
+summary: Check that the experimental hidden dir feature migrates the dir
+
+environment:
+    NAME: test-snapd-tools
+
+prepare: |
+    snap pack "$TESTSLIB"/snaps/"$NAME"
+    "$TESTSTOOLS"/snaps-state install-local "$NAME"
+
+restore: |
+    snap set system experimental.hidden-snap-folder!
+
+execute: |
+    echo "Set experimental hidden snap folder feature"
+    snap set system experimental.hidden-snap-folder=true
+
+    echo "Check that nothing has been migrated yet"
+    snapEnv=$("$NAME".env)
+    echo "$snapEnv" | MATCH "SNAP_USER_DATA=/root/snap/$NAME/x1"
+    echo "$snapEnv" | MATCH "SNAP_USER_COMMON=/root/snap/$NAME/common"
+
+    test -d "$HOME"/snap
+    not test -d "$HOME"/.snap/data
+
+    echo "Take a snapshot"
+    "$NAME".cmd echo "prev_data" > "$HOME"/snap/"$NAME"/current/data
+    snapshot=$(snap save "$NAME" | awk 'FNR == 2 {print $1}')
```

Comment:
I must shamefully admit that I had to google "FNR" for awk. So maybe either a small comment like `First line is the heaer and must be skipped. FNR means line number here` or `snap save test-snapd-tools|tail -1 | awk '{print $1}'` which is worse in every possible way except that it's more familiar to me.